### Question 5
A certain city has two taxi companies: Company A has 80% of the taxis and
Company B has 20% of the taxis. Company A's taxis have a 95% success rate for picking
up passengers on time, while Company B's taxis have a 90% success rate. If a randomly
selected taxi is late, what is the probability that it belongs to Company A

In [1]:
# Probability of selecting a taxi from Company A or Company B
prob_company_a = 0.8
prob_company_b = 0.2

# Probability of a taxi being on time given it's from Company A or Company B
prob_on_time_given_company_a = 0.95
prob_on_time_given_company_b = 0.9

# Probability of a taxi being late given it's from Company A or Company B
prob_late_given_company_a = 1 - prob_on_time_given_company_a
prob_late_given_company_b = 1 - prob_on_time_given_company_b

# Probability of a taxi being late
prob_late = prob_late_given_company_a * prob_company_a + prob_late_given_company_b * prob_company_b

# Probability that a randomly selected late taxi belongs to Company A (using Bayes' Theorem)
prob_company_a_given_late = (prob_late_given_company_a * prob_company_a) / prob_late

print(f"The probability that a randomly selected late taxi belongs to Company A is {prob_company_a_given_late}")

The probability that a randomly selected late taxi belongs to Company A is 0.6666666666666669


The probability that a randomly selected late taxi belongs to Company A is 0.6666666666666669, which means there is approximately a 66.67% chance that a randomly selected late taxi belongs to Company A.